src/ralph/admin/templates/admin/includes/javascripts.html

```diff
@@ -3,27 +3,44 @@
 <script type="text/javascript">window.__admin_utc_offset__ = "{% filter escapejs %}{% now "Z" %}{% endfilter %}";</script>
 <script type="text/javascript">window.__admin_media_prefix__ = "{% static '' %}";</script>
 <script>
-  $(document).ready(function(){
-    $('select.filter').on('change', function(data) {
-        window.location.href = $('option:selected', this).val();
-    });
+$(document).ready(function() {
+      $('#filter-button').bind('click', function() {
+          /**
+           * Clone data filters to filtering form (id: filter-form).
+           * We not used the Django form of change_list because it supports
+           * only the POST request, we need a GET request.
```

Comment:
Explain why we decided to use get request - avoiding auto refreshing? multiple parameters handling? 
This is horrible hack, and needs good explanation for future generations.